dartsim/src/JointFeatures_TEST.cc

```diff
@@ -163,6 +166,467 @@ TEST_F(JointFeaturesFixture, JointSetCommand)
+    world->Step(output, state, input);
+  }
+  EXPECT_LT(pos, joint->GetPosition(0));
+  EXPECT_LT(0, joint->GetVelocity(0));
+}
+
+TEST_F(JointFeaturesFixture, JointSetCombinedLimitsWithForceControl)
+{
+  sdf::Root root;
+  const sdf::Errors errors = root.Load(TEST_WORLD_DIR "test.world");
+  ASSERT_TRUE(errors.empty()) << errors.front();
+
+  auto world = this->engine->ConstructWorld(*root.WorldByIndex(0));
+  auto model = world->GetModel("simple_joint_test");
+  auto joint = model->GetJoint("j1");
+
+  physics::ForwardStep::Output output;
+  physics::ForwardStep::State state;
+  physics::ForwardStep::Input input;
+
+  world->Step(output, state, input);
+
+  auto pos = joint->GetPosition(0);
+
+  joint->SetMinPositionCommand(0, pos - 0.1);
+  joint->SetMaxPositionCommand(0, pos + 0.1);
+  joint->SetMinVelocityCommand(0, -0.25);
+  joint->SetMaxVelocityCommand(0, 0.5);
+  joint->SetMinEffortCommand(0, -1e-6);
+  joint->SetMaxEffortCommand(0, 1e-6);
+
+  for (std::size_t i = 0; i < 100; ++i)
+  {
+    joint->SetForce(0, 100);
+    world->Step(output, state, input);
+  }
+  EXPECT_NEAR(pos, joint->GetPosition(0), 1e-2);
+  EXPECT_NEAR(0, joint->GetVelocity(0), 1e-6);
+
+  for (std::size_t i = 0; i < 100; ++i)
+  {
+    joint->SetForce(0, -100);
+    world->Step(output, state, input);
+  }
+  EXPECT_NEAR(pos, joint->GetPosition(0), 1e-2);
+  EXPECT_NEAR(0, joint->GetVelocity(0), 1e-6);
+
+  joint->SetMinEffortCommand(0, -500);
+  joint->SetMaxEffortCommand(0, 1000);
+
+  for (std::size_t i = 0; i < 100; ++i)
+  {
+    joint->SetForce(0, 1000);
+    world->Step(output, state, input);
+  }
+  // 0.05 because we go 0.1 s with max speed 0.5
+  EXPECT_NEAR(pos + 0.05, joint->GetPosition(0), 1e-2);
+  EXPECT_NEAR(0.5, joint->GetVelocity(0), 1e-6);
+
+  for (std::size_t i = 0; i < 200; ++i)
+  {
+    joint->SetForce(0, 1000);
+    world->Step(output, state, input);
+  }
+  EXPECT_NEAR(pos + 0.1, joint->GetPosition(0), 1e-2);
+  EXPECT_NEAR(0, joint->GetVelocity(0), 1e-6);
+
+  joint->SetPosition(0, pos);
+  EXPECT_NEAR(pos, joint->GetPosition(0), 1e-2);
+
+  joint->SetMinVelocityCommand(0, -1);
+  joint->SetMaxVelocityCommand(0, 1);
+
+  for (std::size_t i = 0; i < 100; ++i)
+  {
+    joint->SetForce(0, 1000);
+    world->Step(output, state, input);
+  }
+  EXPECT_NEAR(pos + 0.1, joint->GetPosition(0), 1e-2);
+  EXPECT_NEAR(1, joint->GetVelocity(0), 1e-6);
+
+  joint->SetPosition(0, pos);
+  EXPECT_NEAR(pos, joint->GetPosition(0), 1e-2);
+
+  joint->SetMinPositionCommand(0, -1e6);
+  joint->SetMaxPositionCommand(0, 1e6);
+
+  for (std::size_t i = 0; i < 100; ++i)
+  {
+    joint->SetForce(0, 1000);
+    world->Step(output, state, input);
+  }
+  EXPECT_LT(pos + 0.1, joint->GetPosition(0));
+  EXPECT_NEAR(1, joint->GetVelocity(0), 1e-6);
+}
+
+// TODO(anyone): position limits do not work very well with velocity control
+// bug https://github.com/dartsim/dart/issues/1583
+#if 0
+TEST_F(JointFeaturesFixture, JointSetPositionLimitsWithVelocityControl)
```

Comment:
Consider disabling the test instead of `#if 0`, so it's at least compiled
```suggestion
TEST_F(JointFeaturesFixture, DISABLED_JointSetPositionLimitsWithVelocityControl)
```

Reply:
I did the requested change